### ISSUE TITLE
Plugin crashes if WS Cleanup plugin is not installed

### ISSUE THREAD

**Issue by halkeye** [v1v/jenkinslint-plugin#3]
I've installed the plugin (I really like the idea of it) but when I goto the JenkinsLint page, I get the table that lists Job Name and JL-1 and such.

How do we get content there? Is a build step need to be added?

**Comment by v1v** [comment]
Hi there,

It is not a job based configuration. It should go through your Jobs and
list all of them. Is your Jenkins setup initialize with some jobs?

Cheers

**Comment by v1v** [comment]
Hi again,

Can you also provide your Jenkins setup (jenkins version, jdk, plugin list,
so on)? then I will reproduce it in my local.

Thanks

**Comment by halkeye** [comment]
Its my real jenkins, so has a bunch of jobs. There is only one job using workflow though, its a good mix and match list.

Um.. Whats the easiest way to get you that information? Do you just want a directory listing of the plugins directory? If you give me your google account email I can give you read access to my server.

**Comment by halkeye** [comment]
Yay looking at logs. I think I spotted the issue (Didn't check logs before opening the ticket sorry)

```
Jul 13, 2015 10:21:06 PM hudson.ExpressionFactory2$JexlExpression evaluate
WARNING: Caught exception evaluating: it.data in /jenkinslint/. Reason: java.lang.reflect.InvocationTargetException
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)

.... Giant Stacktrack ....

        at winstone.BoundedExecutorService$1.run(BoundedExecutorService.java:77)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: hudson/plugins/ws_cleanup/WsCleanup$DescriptorImpl
        at org.jenkins.ci.plugins.jenkinslint.check.CleanupWorkspaceChecker.executeCheck(CleanupWorkspaceChecker.java:22
)
        at org.jenkins.ci.plugins.jenkinslint.JenkinsLintAction.getData(JenkinsLintAction.java:60)
        ... 112 more
```

Plugin List:

```
Enabled   
Name  ↓
Version   
Previously installed version   
Pinned   
Uninstall   

additional-identities-plugin
Additional user property for user identity on other systems. Jenkins then won't create duplicated users when such an identity is extracted during SCM changelog parsing, or other user-related events.
1.0         


Android Emulator Plugin
Starts an Android emulator with given properties before a build, then shuts it down after.
2.13.1  
Downgrade to 2.13


Android Lint Plugin
This plugin parses Android Lint analysis results and visualises the issues found.
2.2         


Ant Plugin
This plugin adds Apache Ant support to Jenkins.
1.2         

Bitbucket OAuth Plugin
This is an authentication plugin for bitbucket.org users
0.4         


Bitbucket Plugin
Integrates with BitBucket
1.1.0   
Downgrade to 1.0


Brakeman Plugin
Parses results from Brakeman, a static-analysis vulnerability scanner for Ruby on Rails.
0.7         


Credentials Plugin
This plugin allows you to store credentials in Jenkins.
1.22    
Downgrade to 1.22
Unpin


CVS Plug-in
Integrates Jenkins with CVS version control system using a modified version of the Netbeans cvsclient.
2.12    
Downgrade to 2.11
Unpin


Docker build step plugin
This plugin allows to add various docker commands to your job as build steps.
1.24            


DocLinks plugin
This plugin allows you to publish documents that are created in the build step.
0.6         


Durable Task Plugin
Library offering an extension point for processes which can run outside of Jenkins yet be monitored.
1.5 
Downgrade to 1.5


Emma plugin
This plugin integrates EMMA code coverage reports to Hudson.
1.29            


Environment File Plugin
This plugin can be used to set environment variables from a file.
1.2         


Environment Injector Plugin
This plugin makes it possible to set an environment for the builds. 

1.91.2  
Downgrade to 1.91.1


External Monitor Job Type Plugin
Adds the ability to monitor the result of externally executed jobs.
1.4         

File To Env Plugin
This plugin gives the ability to copy the contents of a file into an environmental variable
1.0-SNAPSHOT (private-12/16/2014 00:54-jenkins)         


FindBugs Plug-in
This plug-in collects the FindBugs analysis results of the project modules and visualizes the found warnings.
If you like this open source plug-in please consider supporting my work by buying my Android game Inca Trails.

4.60    
Downgrade to 4.59


Gem Publisher Plugin
This plugin publishes rubygems to rubygems.org using their API.
1.0         


GIT client plugin
Shared library plugin for other Git related Jenkins plugins.
1.17.1  
Downgrade to 1.16.1


GIT plugin
This plugin integrates GIT with Jenkins.
2.3.5   
Downgrade to 2.3.4


Git server plugin
Allows Jenkins to act as a Git server.
1.6 
Downgrade to 1.5


GitHub API Plugin
This plugin provides GitHub API for other plugins.
1.67    
Downgrade to 1.67


GitHub plugin
This plugin integrates GitHub to Jenkins.
1.11.3  
Downgrade to 1.11


Go Plugin
Automatically installs and sets up the Go programming language (golang) tools for a build.
1.1         


Google Login Plugin
Allows you to login to Jenkins with a Google account and restrict access to a Google Apps Domain.
1.1         


Gradle plugin
This plugin allows Jenkins to invoke Gradle build scripts directly.
1.24            


graphiteIntegrator
This plugin allows you to send these metrics : number of tests, tests skipped, tests failed, build duration, cobertura total line coverage and cobertura total branch coverage to one or more graphite servers. If you don't have a graphite server you can use : https://www.hostedgraphite.com to test. For cobertura metrics you need to install cobertura plugin and run cobertura:cobertura in goals section. Be sure to run jenkins in a Jdk 7, because the plugin only works with this version of jdk.
1.2 
Downgrade to 1.1


Heroku Plugin for Jenkins
A plugin for interacting with Heroku during Jenkins builds.
0.7.1-BETA          


HTML5 Notifier Plugin
The HTML5 Notifier Plugin provides W3C Web Notifications support for builds.
1.5 
Downgrade to 1.3


Hudson Ruby Plugin
This plugin will let users use Ruby in the build scripts.
1.2         


Hudson SCP publisher plugin
This plugin uploads build artifacts to repository sites using SCP (SSH) protocol.
1.8         


JaCoCo plugin
This plugin allows you to capture code coverage report from JaCoCo. Jenkins will generate the trend report of coverage. This plugin is fork of the [Emma Plugin]. Big part of the code structure comes from it, however, it is completely refactored. It also includes functionality similar to the [Emma Coverage Column] which allows to include a column in Dashboards which displays the latest overall coverage numbers and links to the coverage report.
1.0.19  
Downgrade to 1.0.18


Javadoc Plugin
This plugin adds Javadoc support to Jenkins.
1.3 
Downgrade to 1.2
Unpin


JenkinsLint Plugin
The JenkinsLint Plugin.
0.1.0           


JUnit Plugin
Allows JUnit-format test results to be published.
1.6 
Downgrade to 1.5
Unpin


LDAP Plugin
Adds LDAP authentication to Jenkins
1.11    
Downgrade to 1.6
Unpin


Mailer Plugin
This plugin allows you to configure email notifications. This is a break-out of the original core based email component.
1.15    
Downgrade to 1.12
Unpin


MapDB API Plugin
This plugin provides a shared dependency on the MapDB library so that other plugins can co-operate when using this library.
1.0.6.0         


Matrix Authorization Strategy Plugin
Offers matrix-based security authorization strategies (global and per-project).
1.2 
Downgrade to 1.1
Unpin


Matrix Project Plugin
Multi-configuration (matrix) project type.
1.4.1   
Downgrade to 1.3

Maven Integration plugin
Jenkins plugin for building Maven 2/3 jobs via a special project type.
2.9 
Downgrade to 2.8
Unpin


Metrics Graphite Reporting Plugin
This plugin streams Metrics to a Graphite server.
3.0.0           


Metrics Plugin
This plugin exposes the Metrics API to Jenkins plugins.
3.0.10  
Downgrade to 3.0.9


Modern Status
A modern status icon set for Jenkins.
1.2 
Downgrade to 1.0


MSBuild Plugin
This plugin makes it possible to build a Visual Studio project (.proj) and solution files (.sln).
1.24            


MSTest plugin
This plugin converts MSTest TRX test reports into JUnit XML reports so it can be integrated with Hudson's JUnit features.
0.17    
Downgrade to 0.16


MSTestRunner plugin
This plugin run MSTest command line tool to execute unit tests for .NET projects.
1.1.2           


Notification plugin
This plugin from Tikal Knowledge allows sending running Jobs status notifications.
1.9         


Nuget Plugin
This plugin is to trigger a build when a NuGet dependency has an updated version.
0.4         


nvm wrapper
No description available.
0.0.1-SNAPSHOT (private-10/18/2014 16:01-jenkins)           


OWASP Markup Formatter Plugin
Uses the OWASP Java HTML Sanitizer to allow safe-seeming HTML markup to be entered in project descriptions and the like.
1.3 
Downgrade to 1.1
Unpin


PAM Authentication plugin
Adds Unix Pluggable Authentication Module (PAM) support to Jenkins.
1.2 
Downgrade to 1.1
Unpin


PegDown Formatter Plugin
Format descriptions using Markdown syntax. Use simple wiki-like markup instead of HTML.
1.3 
Downgrade to 1.3


Port Allocator Plug-in
This plugin allocates free ports as environment variables.
1.8         


Project Description Setter
Set the project description from a file in the workspace (à la GitHub README.md)
1.1         


promoted builds plugin
This plugin implements a "promoted build" feature where a build of one job can be marked as "promoted" when it passes certain criteria.
2.21    
Downgrade to 2.20


Publish Over SSH
Send build artifacts over SSH
1.12            


Rake plugin
The Rake plugin allows to execute rake task from Jenkins.
1.8.0   
Downgrade to 1.8.0


ruby-runtime
Provides the Ruby runtime and bindings required to implement plugins in Ruby.
0.12            


RubyMetrics plugin for Jenkins
This plugin integrates a bunch of ruby coverage tools (Rcov, Saikuro, Rails stats...) to Jenkins. Currently it just supports Rcov reports.
1.6.3   
Downgrade to 1.6.2


Run Condition Plugin
Core conditions to select whether to execute a build step or publisher. Used by the [Flexible Publish Plugin] and the [Conditional BuildStep Plugin].
1.0         


Rvm
This plugin runs your jobs in the RVM managed ruby+gemset of your choice
0.4         


SCM API Plugin
This plugin provides a new enhanced API for interacting with SCM systems.
0.2         


Script Security Plugin
Allows Jenkins administrators to control what in-process scripts can be run by less-privileged users.
1.13    
Downgrade to 1.13

Secure Requester Whitelist Plugin
Allows an administrator to specify sites trusted to make JSONP or primitive-XPath REST API requests.
1.0         


Slack Notification Plugin
This plugin is a Slack notifier that can publish build status to Slack channels.
1.7 
Downgrade to 1.7-SNAPSHOT (private-12/09/2014 21:21-jenkins)


SSH Agent Plugin
This plugin allows you to provide SSH credentials to builds via a ssh-agent in Jenkins.
1.6 
Downgrade to 1.5


SSH Credentials Plugin
This plugin allows you to store SSH credentials in Jenkins.
1.11    
Downgrade to 1.10
Unpin


SSH Slaves plugin
This plugin allows you to manage slaves running on \*nix machines over SSH.
1.9         

Static Analysis Collector Plug-in
This plug-in is an add-on for the plug-ins Checkstyle, Dry, FindBugs, PMD, Tasks, and Warnings: the plug-in collects the different analysis results and shows the results in a combined trend graph. Additionally, the plug-in provides health reporting and build stability based on these combined results.
If you like this open source plug-in please consider supporting my work by buying my Android game Inca Trails.

1.43    
Downgrade to 1.42


Static Analysis Utilities
This plug-in provides utilities for the static code analysis plug-ins.
If you like this open source plug-in please consider supporting my work by buying my Android game Inca Trails.

1.71    
Downgrade to 1.71


Subversion Plug-in
This plugin adds the Subversion support (via SVNKit) to Jenkins.
2.5 
Downgrade to 2.5
Unpin


ThinBackup
This plugin simply backs up the global and job specific configurations (not the archive or the workspace).
1.7.4           


Token Macro Plugin
This plug-in adds reusable macro expansion capability for other plug-ins to use.
1.10            


Translation Assistance plugin
This plugin adds an additional dialog box in every page, which enables people to contribute localizations for the messages they are seeing in the current page.
1.12    
Downgrade to 1.10
Unpin


Travis YML Plugin
This plugin runs your jobs using .travis.yml file in your project.
0.1.0           


Violations plugin
This plugin does reports on checkstyle, csslint, pmd, cpd, fxcop, pylint, jcReport, findbugs, and perlcritic violations.
0.7.11          


VSTest Runner plugin
This plugin run 'VSTest.Console.exe' command line tool to execute unit tests for .NET projects.
1.0.4           


Windows Slaves Plugin
Allows you to connect to Windows machines and start slave agents on them.
1.0         

Workflow: Aggregator
Collects all workflow-related plugins as dependencies to make them easier to install and demonstrate.
1.6 
Downgrade to 1.5


Workflow: API
Plugin that defines workflow API.
1.6 
Downgrade to 1.6


Workflow: Basic Steps
Commonly used steps for workflows.
1.6 
Downgrade to 1.5


Workflow: Durable Task Step
Workflow steps running external processes that may survive a Jenkins restart or slave reconnection.
1.6 
Downgrade to 1.5


Workflow: Execution Support
Common utility implementations to build workflow plugin
1.6 
Downgrade to 1.5


Workflow: Global Shared Library for CPS workflow
Global shared library for workflow scripts
1.6 
Downgrade to 1.6


Workflow: Groovy CPS Execution
Workflow execution engine based on continuation passing style transformation of Groovy scripts.
1.6 
Downgrade to 1.5


Workflow: Job
Defines a new job type for workflows and provides their generic user interface.
1.6 
Downgrade to 1.5


Workflow: SCM Step
Adds workflow steps to check out or update working sources from various SCMs (version control).
1.6 
Downgrade to 1.6


Workflow: Step API
API for asynchronous build step primitive.
1.6 
Downgrade to 1.6


Xvfb plugin
This plugin starts before build, and stops after the build the Xvfb virtual framebuffer X11 server.
```

**Comment by v1v** [comment]
I've just seen where the issue is:

org.jenkins.ci.plugins.jenkinslint.check.CleanupWorkspaceChecker

Wscleanup plugin is optional, but for some reason wasn't installed. I will
patch that issue.

Thanks for the spot

Cheers

**Comment by v1v** [comment]
Workaround:
- Install WS Cleanup plugin (https://wiki.jenkins-ci.org/display/JENKINS/Workspace+Cleanup+Plugin)

Meanwhile I'll put that fix in place to avoid those runtime exceptions:
- https://issues.jenkins-ci.org/browse/JENKINS-29418

Raised a Jira ticket to track those issues too

Cheers

**Comment by v1v** [comment]
Already released another version (0.1.1) It does fix that dependency and it does include another check (hardcoded shell/batch scripts).

It should be in the update center shortly.

Closing this ticket now. Please let me know if you still have issues

Cheers
